### ISSUE TITLE
GCE windows file upload fails

### ISSUE THREAD

**Issue by JorritSalverda** [mitchellh/packer#2786]
I get the following error over and over on gce windows server 2012 r2. It seems to have to do with the source image being a sysprepped image already, because it never happens from the google base image.

I had a fix that worked for a while, restart the server before uploading the file. But that stopped working now as well.

```
09:45:18.205 1443519918,,ui,say,==> googlecompute: Connected to WinRM!
09:45:18.207 1443519918,,ui,say,==> googlecompute: Uploading ./packer-templates/sysprep.ps1 => c:/sysprep.ps1
09:45:24.146 1443519924,,ui,error,==> googlecompute: Upload failed: Error uploading file to 
$env:TEMP\winrmcp-560a5dae-a85b-b321-21cc-59cd25e16b60.tmp: Couldn't create shell: 
unknown error Post http://104.155.64.77:5985/wsman: read tcp 
172.17.0.2:45896->104.155.64.77:5985: read: connection reset by peer
```

The AWS build that runs in parallel never has these issues.

**Comment by JorritSalverda** [comment]
By adding a pointless inline powershell provisioner in front of it it seems more stable:

```
  "provisioners":
  [
    {
      "type": "powershell",
      "inline": ["dir c:\\"]
    },
    {
      "type": "file",
      "source": "./packer-templates/sysprep.ps1",
      "destination": "c:/sysprep.ps1"
    },  
```

Do the file and powershell provisioner have a different way of waiting for the machine to be ready?

**Comment by cbednarski** [comment]
Thanks for the report! It sounds like there is some specific permutation here that causes this particular problem but it sounds to me like there is more than one bug here. I'll need to do some digging.

**Comment by JorritSalverda** [comment]
This happens again every now and then, perhaps because the windows vm on google is slightly slower to start up at times? Seems to be gone again after a couple of days, but makes it impossible for us to bake images during that time.

In this case it happens for uploading a 6.4MB file that worked fine. A couple of other small files upload without any issues.

```
==> googlecompute: Upload failed: Error uploading file to $env:TEMP\winrmcp-57bedb56-da78-4052-962f-c28c3490bd1b.tmp: http error: 500 - 

<s:Envelope xml:lang="en-US" xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:x="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:e="http://schemas.xmlsoap.org/ws/2004/08/eventing" xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd">
  <s:Header>
    <a:Action>http://schemas.dmtf.org/wbem/wsman/1/wsman/fault</a:Action>
    <a:MessageID>uuid:EC0138CF-C5C1-47A1-B735-4BF987DBF713</a:MessageID>
    <a:To>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:To>
    <a:RelatesTo>uuid:5204b563-f8ac-48e6-6064-3665b08c8f9c</a:RelatesTo>
  </s:Header>
  <s:Body>
    <s:Fault>
      <s:Code>
        <s:Value>s:Sender</s:Value>
        <s:Subcode>
          <s:Value>w:InvalidSelectors</s:Value>
        </s:Subcode>
      </s:Code>
      <s:Reason>
        <s:Text xml:lang="en-US">The WS-Management service cannot process the request because the request contained invalid selectors for the resource. </s:Text>
      </s:Reason>
      <s:Detail>
        <w:FaultDetail>http://schemas.dmtf.org/wbem/wsman/1/wsman/faultDetail/UnexpectedSelectors</w:FaultDetail>
        <f:WSManFault xmlns:f="http://schemas.microsoft.com/wbem/wsman/1/wsmanfault" Code="2150858843" Machine="192.158.28.128">
          <f:Message>The request for the Windows Remote Shell with ShellId 78F3947D-928C-4ECF-9046-395687BCAD33 failed because the shell was not found on the server. Possible causes are: the specified ShellId is incorrect or the shell no longer exists on the server. Provide the correct ShellId or create a new shell and retry the operation. </f:Message>
        </f:WSManFault>
      </s:Detail>
    </s:Fault>
  </s:Body>
</s:Envelope>
```

**Comment by mwhooker** [comment]
There have been a bunch of changes to packer since this was opened. If you're still seeing these errors, please open another ticket
